### PR TITLE
[BEAM-14451] Support export to BigQuery in FhirIO.Export

### PR DESCRIPTION
**Pull request by lnogueir**
Currently, the `FhirIO.Export` transform only supports exports of Fhir resources to Google Cloud Storage. This PR expands its functionality to also be able to export to Google BigQuery by introducing a new method, namely `exportFhirResourceToBigQuery`, to `HealthcareApiClient` and modifying the existing `DoFn` `@ProcessElement` method to call the appropriate export method based on the input URI.

Also, changed the export behavior to not return a `PCollection` of the exported resources and instead output the URI where the store was exported to. @msbukal and I agreed that this would be a better approach because it gives more flexibility to the user by allowing them to chain with transforms tailored for reading resources.

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [x] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/#make-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by lnogueir** [comment]
Run Java PreCommit

**Comment by lnogueir** [comment]
R: @msbukal

**Comment by kennknowles** [comment]
@lnogueir do you have a Jira account so I can assign it to you?

**Comment by chamikaramj** [comment]
cc: @pabloem

**Comment by lnogueir** [comment]
> @lnogueir do you have a Jira account so I can assign it to you?

Yes, I do have a Jira account, it's `lnogueir`. I tried emailing [dev@beam.apache.org](mailto: dev@beam.apache.org) requesting to be a contributor on Jira, but my google email currently doesn't allow me to email external domains.
If you could do that, it would be great!

File: sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/healthcare/FhirIOLROIT.java [hunk 1]
```diff
@@ -38,6 +38,7 @@ public class FhirIOLROIT {
 
   private transient HealthcareApiClient client;
   private final String project;
+  private static final String BQ_DATASET_ID = "big_query_storage";
```
Comment (lnogueir) [inline sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/healthcare/FhirIOLROIT.java:41]: I saw that BigQueryIO uses this dataset id for some of their tests, so figured we could use it here. However, I get permission denied when trying to export there. Should we request access to this dataset or should we create our own? @msbukal @pabloem

File: sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/healthcare/FhirIOTest.java [hunk 2]
```diff
@@ -146,4 +151,17 @@ public void test_FhirIO_failedPatientEverything() {
     PAssert.that(everythingResult.getPatientCompartments()).empty();
     pipeline.run();
   }
+
+  @Test
+  public void test_FhirIO_Export_invalidUri() {
+    final String invalidUri = "someInvalidUri";
+    pipeline
+        .apply(Create.of(Arrays.asList("resource1", "resource2")))
```
Comment (msbukal) [inline sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/healthcare/FhirIOTest.java:159]: What's the point of this line? Also just call FhirIO.exportResources(...) same as the other tests

**Review by msbukal — APPROVED** [review]
LGTM with some minor comments

**Comment by pabloem** [comment]
I'll take a look this week

**Comment by pabloem** [comment]
LGTM!

**Pull request merged**